### DISCUSSION TITLE
Falscher Wert für Luftfeuchtigkeit in HM-Heizgruppen

### DISCUSSION THREAD

**Discussion by mipri25** [OpenCCU/OpenCCU#2274] [🇩🇪 Fragen&Antworten]
Hallo,

ich habe hier zwei HM-Heizgruppen, und in beiden wird im seit Version 3.67.10.20230114 vorhandenen Feld für die Luftfeuchtigkeit im virtuellen Gruppengerät ein falscher Wert angezeigt, nämlich stets 20%.

Gruppe 1 besteht aus einem Wandthermostat HM-TC-IT-WM-W-EU und zwei Heizkörperthermostaten HM-CC-RT-DN. Im virtuellen Gruppengerät wird die Luftfeuchtigkeit stets mit 20% angegeben, während der für den Wandthermostat angezeigte Wert korrekt ist. Die Raumtemperatur und die Solltemperatur werden dagegen auch im Gruppengerät korrekt angezeigt, und die Heizgruppe funktioniert auch ansonsten einwandfrei, d.h. eine veränderte Solltemperatur an einem Gerät wird an die jeweils anderen übermittelt, und die Heizkörperthermostate übernehmen die Raumtemperatur vom Wandthermostat.

Gruppe 2 besteht lediglich aus einem HM-CC-RT-DN und einem Fensterkontakt HM-Sec-SCo. Obwohl hier also gar keine Luftfeuchtigkeit gemessen wird, wird sie im virtuellen Gruppengerät trotzdem mit 20% angezeigt.

Hardware ist ein ELV Charly mit Raspberrymatic 3.67.10.20230225.

Zur Kontrolle habe ich noch einen Versuch unter VirtualBox unternommen, mit derselben Version der Raspberrymatic. Hier habe ich eine leere HM-Heizgruppe ohne Geräte angelegt, und selbst für diese wird die Luftfeuchtigkeit mit 20% angezeigt.

Tritt dieses Problem nur bei mir auf, und falls ja, was könnte die Ursache sein?
Oder ist es ein allgemeines Problem? Dann würde ich einen Issue daraus machen.

Besten Dank schonmal!

**Comment by udochrist** [comment]
Das sieht nach einem Fehler aus. Allerdings bei mir anders als bei dir.
Ich habe eine Gruppe ohne Wandthermostat und mehrere Gruppen mit Wandthermostat.

Eigentlich sollte in der einen Gruppe kein Luftfeuchtigkeit-Wert vorliegen da den kein Gerät liefern kann. In dieser Gruppe wird aber 20% angezeigt.
Die anderen Gruppen zeigen bei mir allerdings die korrekten werte an.

Eventuell kommt der echte werte deines Thermostats nicht in der Gruppe an und 20% ist der default - wert.

**Reply by mipri25** [reply]
Da es also offensichtlich auch Heizgruppen gibt, bei denen die Anzeige der Luftfeuchtigkeit funktioniert, habe ich mich nochmal auf die Suche nach der Ursache des Problems gemacht, und nachdem hier nichts zu finden war, habe ich schließlich die ganze Zentrale neu gestartet. Seitdem wird die Luftfeuchtigkeit in der Gruppe mit dem Wandthermostat korrekt angezeigt (die anderen zeigen nach wie vor konstant 20% an).

Falls es sich also um ein allgemeines Problem handelt, dann tritt es wohl nur vom Anlegen einer neuen HM-Heizgruppe bis zum nächsten Neustart der Zentrale auf.

**Comment by jp112sdl** [comment]
Das liegt an der Idee von hier https://github.com/jens-maus/RaspberryMatic/discussions/2093 und der Umsetzung von mir hier https://github.com/jens-maus/RaspberryMatic/pull/2106

Es ist wohl so, dass ohne Wandthermostat von der Gruppe aus dem HMIPServer immer der MIN-Value zurückgeliefert wird.

@jens-maus Soll ich den 0042-WebUI-HeatingControl-HumidityBatteryValveState.patch dahingehend ändern, dass das Feld nur angezeigt wird, wenn die Luftfeuchtigkeit > 20% ist?

**Reply by MichaelN0815** [reply]
Ich würde auf == 20 prüfen und dann "n/a" anzeigen. So ist das Layout wenigstens immer gleich.

**Reply by jens-maus** [reply]
Und was ist wenn die Luftfeuchtigkeit wirklich 20% ist? Ist zwar recht trocken, aber kann ja vorkommen…

**Reply by jens-maus** [reply]
> @jens-maus Soll ich den 0042-WebUI-HeatingControl-HumidityBatteryValveState.patch dahingehend ändern, dass das Feld nur angezeigt wird, wenn die Luftfeuchtigkeit > 20% ist?

Hab mir den Patch nochmal grob angeschaut. Denke in der Tat das ne Anpassung wohl sinnvoll wäre wenn da kein 20% angezeigt werden soll, was ja darauf deutet das der Wert nicht valide ist. Ich würde auch dafür plädieren hier "n/a" anzuzeigen damit klarer ist das der Wert aktuell nicht existiert/valide ist.

**Reply by jp112sdl** [reply]
> Und was ist wenn die Luftfeuchtigkeit wirklich 20% ist? 

Dann muss im HMIPServer nachgebessert werden, so dass bei einem virtuellen Gruppengerät bei Fehlen eines Wandthermostats _oder eines anderen Sensors, der Luftfeuchtigkeit liefert_, dann nicht fälschlicherweise 20% ausgegeben werden.

Keine Ahnung, ob man das sogar als Bug bezeichnen könnte.
Denn der Datenpunkt `ACTUAL_HUMIDITY` eines virtuellen `INT000...`-Gruppengeräts  ist bei Fehlen eines o.g. Geräts dennoch vorhanden und liefert immer 20%

Ist das bei einer virtuellen HmIP-Gruppe eigentlich auch so? Kann das mal jemand testen?

**Reply by jp112sdl** [reply]
>  Ich würde auch dafür plädieren hier "n/a" anzuzeigen 

@jens-maus: Oben oder Unten ?

<img width="231" alt="Bildschirmfoto 2023-04-02 um 14 51 48" src="https://user-images.githubusercontent.com/18190139/229353946-0f2c9694-f081-4cdb-a27c-7972e2a7318b.png">
<img width="226" alt="Bildschirmfoto 2023-04-02 um 14 52 27" src="https://user-images.githubusercontent.com/18190139/229353983-0e0a890f-49e5-4c60-b500-700642b7c907.png">

**Reply by jens-maus** [reply]
Oben, aber als lowercase "n/a" ;)
